Data Preparation.R

```diff
@@ -0,0 +1,173 @@
+
+
+## 2. SMR50 extract ##
+Query_SMR50 <- paste0("select link_no, gls_cis_marker, admission_date, 
+                       discharge_date, sex, ",
+                      "inpatient_daycase_identifier from ANALYSIS.SMR01_1E_PI ",
+                      "where discharge_date >= {d '", start_date, "'} 
+                       AND discharge_date <= {d '", end_date, "'} ",
+                      "ORDER BY link_no, cis_marker, admission_date, 
+                       discharge_date, admission, discharge, uri")
+
+data_50  <- as_tibble(dbGetQuery(SMRA, Query_SMR50)) %>%
+            mutate(recid = "SMR50") %>% clean_names()
+
+
+## 3. SMR04 extract ##
+Query_SMR04 <- paste("select link_no, cis_marker, admission_date, 
+                      discharge_date, sex,",
+                     "management_of_patient from ANALYSIS.SMR04_PI ",
+                     "WHERE management_of_patient IN ('1', '3', '5', '7', 'A')",
+                     "ORDER BY link_no, cis_marker, admission_date, 
+                      discharge_date, admission, discharge, uri")
+
+data_04 <- as_tibble(dbGetQuery(SMRA, Query_SMR04)) 
+data_04 %<>% filter(DISCHARGE_DATE >= "2009-10-01" & 
+                    DISCHARGE_DATE < "2018-03-31") %>% drop_na() %>%
+                    mutate(recid = "SMR04") %>% clean_names()
+
+
+## 4. Deaths extract ##
+Query_Deaths <- paste0("select link_no, date_of_death, institution, sex, ",
+                       "underlying_cause_of_death, cause_of_death_code_0, ",
+                       "cause_of_death_code_1, cause_of_death_code_2, ",
+                       "cause_of_death_code_3, cause_of_death_code_4, ",
+                       "cause_of_death_code_5, cause_of_death_code_6, ",
+                       "cause_of_death_code_7, cause_of_death_code_8, ",
+                       "cause_of_death_code_9 ",
+                       "from ANALYSIS.GRO_DEATHS_C ",
+                       "where date_of_death >= {d '", deaths_start_date, "'}")
+
+data_deaths  <- as_tibble(dbGetQuery(SMRA, Query_Deaths)) %>%
+                mutate(recid = "Deaths") %>% clean_names()
+
+
+#######################################
+## SECTION 3 - Data preparation      ##
+#######################################
+
+
+# 1. Create SMR activity file (SMR01_50)
+
+smr01_50 <- rbind(data_01,data_50) %>% 
+  
+# 2. Select only inpatient_day case
+  
+filter(inpatient_daycase_identifier == "I") %>% 
+  
+# 3. Aggregate to CIS level. Record first admission and last discharge dates.
+  
+group_by(link_no,gls_cis_marker) %>% 
+                                 summarise(admission_date = min(admission_date), 
+                                 discharge_date = max(discharge_date)) %>% 
+                                 ungroup() 
+
+### Duplicates Removal ###
+
+#### SMR01_50 ####
+
+# Aggregate to link_no and gls_cis_marker to get rid of duplicated rows. 
+smr01_50_final <- smr01_50 %>% group_by(link_no, gls_cis_marker) 
+                            
+#### SMR04 ####
```

Comment:
You are quite inconsistent with comment formatting. Can you please follow the style guide with this, and also more generally? The style guide can be found here: https://github.com/Health-SocialCare-Scotland/R-Resources/blob/master/PHI%20R%20style%20guide.md

There is also a template script with an example of how they should be formatted here: https://github.com/Health-SocialCare-Scotland/r-project-structure/blob/master/code/code.R